daemon/containerd/image_list.go

```diff
@@ -77,29 +78,48 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 			continue
 		}
 
-		platforms, err := images.Platforms(ctx, contentStore, img.Target)
-		if err != nil {
-			return nil, err
-		}
-
-		for _, platform := range platforms {
-			image, chainIDs, err := i.singlePlatformImage(ctx, contentStore, img, platform)
-			if err != nil {
-				return nil, err
-			}
-			if image == nil {
-				continue
+		err := images.Walk(ctx, images.HandlerFunc(func(ctx context.Context, desc v1.Descriptor) ([]v1.Descriptor, error) {
+			if images.IsIndexType(desc.MediaType) {
+				return images.Children(ctx, contentStore, desc)
 			}
 
-			summaries = append(summaries, image)
+			if images.IsManifestType(desc.MediaType) {
+
+				// Ignore buildkit attestation manifests
+				// https://github.com/moby/buildkit/blob/master/docs/attestations/attestation-storage.md
+				if _, has := desc.Annotations[attestation.DockerAnnotationReferenceType]; has {
+					return nil, nil
+				}
+
+				if desc.Platform == nil {
+					return nil, nil
+				}
```

Comment:
Is this correct? 🤔 ISTR `Platform` to be optional (as originally it wasn't there, so to be interpreted as `linux/amd64`). ISTR this was the reason to use `unknown/unknown` as platform for these attestations

https://github.com/opencontainers/image-spec/blob/main/image-index.md

> platform object
>
> This OPTIONAL property describes the minimum runtime requirements of the image. This property SHOULD be present if its target is platform-specific.

Reply:
Yeah it's not really correct, handled it by reading the platform from the image config.